plugins/by-name/luasnip/default.nix

```
@@ -39,54 +33,265 @@ let
             List of paths to load.
           '';
 
-      exclude = helpers.mkNullOrOption (with lib.types; maybeRaw (listOf (maybeRaw str))) ''
+      exclude = lib.nixvim.mkNullOrOption (with lib.types; maybeRaw (listOf (maybeRaw str))) ''
         List of languages to exclude, by default is empty.
       '';
 
-      include = helpers.mkNullOrOption (with lib.types; maybeRaw (listOf (maybeRaw str))) ''
+      include = lib.nixvim.mkNullOrOption (with lib.types; maybeRaw (listOf (maybeRaw str))) ''
         List of languages to include, by default is not set.
       '';
     };
   };
 in
-{
-  imports =
-    let
-      basePluginPath = [
-        "plugins"
-        "luasnip"
-      ];
-    in
-    [
-      # TODO introduced 2024-08-04. Remove after 24.11
-      (lib.mkRenamedOptionModule (basePluginPath ++ [ "extraConfig" ]) (basePluginPath ++ [ "settings" ]))
-    ];
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "luasnip";
+  package = "luasnip";
+  setup = ".config.setup";
 
-  options.plugins.luasnip = {
-    enable = mkEnableOption "luasnip";
+  maintainers = [ lib.maintainers.HeitorAugustoLN ];
 
-    package = lib.mkPackageOption pkgs "luasnip" {
-      default = [
-        "vimPlugins"
-        "luasnip"
-      ];
-    };
+  settingsOptions = {
+    keep_roots = defaultNullOpts.mkBool false ''
+      Whether snippet-roots should be linked.
+    '';
 
-    settings = mkOption {
-      type = with types; attrsOf anything;
-      description = ''
-        Options provided to the `require('luasnip').config.setup()` function.",
-      '';
-      example = {
-        enable_autosnippets = true;
-        store_selection_keys = "<Tab>";
-      };
-      default = { };
-    };
+    link_roots = defaultNullOpts.mkBool false ''
+      Whether snippet-roots should be linked.
+    '';
+
+    exit_roots = defaultNullOpts.mkBool true ''
+      Whether snippet-roots should exit at reaching at their last node, $0.
+      This setting is only valid for root snippets, not child snippets.
+      This setting may avoid unexpected behavior by disallowing to jump earlier (finished) snippets.
+    '';
+
+    link_children = defaultNullOpts.mkBool false ''
+      Whether children should be linked.
+    '';
+
+    update_events =
+      defaultNullOpts.mkNullableWithRaw (with types; either str (listOf str)) "InsertLeave"
+        ''
+          Choose which events trigger an update of the active nodes' dependents.
+        '';
+
+    region_check_events =
+      defaultNullOpts.mkNullableWithRaw (with types; either str (listOf str)) { __raw = "nil"; }
```

```suggestion
      defaultNullOpts.mkNullableWithRaw (with types; either str (listOf str)) (literalLua "nil")
```

Going forward, we should aim to use `lib.nixvim.litearalLua` instead of `__raw` in examples & defaultText.

---

Should be fixed now